app/models/trade/annual_report_upload.rb

```diff
@@ -84,8 +84,56 @@ def process_validation_rules
     end
   end
 
-  def submit
-    # TODO
+  def run_primary_validations
+    @validation_errors = run_validations(
+      Trade::ValidationRule.where(:is_primary => true)
+    )
+  end
+
+  def submit(submitter)
+    run_primary_validations
+    unless @validation_errors.count == 0
+      self.errors[:base] << "Submit failed, primary validation errors present."
+      return false
+    end
+
+    return false unless sandbox.copy_from_sandbox_to_shipments(submitter)
+    # remove uploaded file
```

Comment:
Actually on second thoughts - I think we can skip that for now. If in the future we decide to let EPIX users upload CSV files, this becomes relevant again.